test/hack/resource/clean/main.go

```diff
@@ -73,10 +76,10 @@ func main() {
 		resourceLogger := logger.With("type", resourceTypes[i].String())
 		var ids []string
 		var err error
-		if clusterName == "" {
+		if lo.FromPtr(clusterName) == "" {
 			ids, err = resourceTypes[i].GetExpired(ctx, expirationTime)
 		} else {
```

Comment:
Can you leave a comment here so it's clearer what's happening? Maybe it was just tougher to figure out what was going on because I had to click through a couple of links not in the PR to find what this was doing. 

i.e. If there's no cluster defined, clean up all resources. otherwise, only cleanup the resources associated with the cluster.